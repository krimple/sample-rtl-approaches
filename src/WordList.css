.wordContainer {
    display: -webkit-flex;
    display: flex;
    -webkit-align-items: center;
    align-items: initial;
    justify-content: left;
    -webkit-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: stretch;
    align-content: stretch;
    flex-grow: 0; /* do not grow   - initial value: 0 */
    flex-shrink: 0; /* do not shrink - initial value: 1 */
    flex-basis: 200px;
}

.wordList {
    text-align: left;
    color: orange;
    height: 200px;
    border-top: 2px solid blue;
    border-bottom: 2px solid blue;
    margin: 25px;
    padding: 10px;
    font-size: 25px;
}

