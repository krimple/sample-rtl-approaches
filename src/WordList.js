import PropType from 'prop-types';
import './WordList.css';

/**
 * WordList - renders the list of words added by the player
 * @param wordList the list of words to render
 *
 * Note - we are using `data-testid` for our elements to be able to find them using
 * testing-libary/react and the `getByTestId` locator - see the test.
 * A better way to do this is to potentially add screen reader or aria attributes
 * that properly render the content for selection normally via the DOM / HTML elements.
 */
export default function WordList ({ wordList }) {

  return (
    <div className="wordContainer wordList">
      {
        wordList.map((word, index) => {
          return (
            <div key={'word-' + index} data-testid={index}>
              <i className="bi bi-plus-lg"></i>
              {word}
            </div>
          );
        })
      }
    </div>
  );
}

WordList.propTypes = {
  wordList: PropType.arrayOf(PropType.string)
};

