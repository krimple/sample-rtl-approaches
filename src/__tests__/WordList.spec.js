import { render, screen } from '@testing-library/react';
import WordList from '../WordList';

describe('<WordList />', () => {
  it('matches snapshot with empty wordlist', () => {
    const { asFragment } = render(<WordList wordList={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // TODO - figure out better pattern
  it('matches snapshot with 3 words', () => {
    const { asFragment } = render(<WordList wordList={['Foo', 'Bar', 'Baz']} />);
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByTestId('0')).not.toBeNull();
    expect(screen.getByTestId('1')).not.toBeNull();
    expect(screen.getByTestId('2')).not.toBeNull();
  });
});

