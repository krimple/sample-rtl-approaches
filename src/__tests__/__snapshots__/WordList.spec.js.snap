// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordList /> matches snapshot with 3 words 1`] = `
<DocumentFragment>
  <div
    class="wordContainer wordList"
  >
    <div
      data-testid="0"
    >
      <span
        class="wordlist-bullet"
      >
        *
      </span>
      Foo
    </div>
    <div
      data-testid="1"
    >
      <span
        class="wordlist-bullet"
      >
        *
      </span>
      Bar
    </div>
    <div
      data-testid="2"
    >
      <span
        class="wordlist-bullet"
      >
        *
      </span>
      Baz
    </div>
  </div>
</DocumentFragment>
`;

exports[`<WordList /> matches snapshot with empty wordlist 1`] = `
<DocumentFragment>
  <div
    class="wordContainer wordList"
  />
</DocumentFragment>
`;
